app/controllers/walks_controller.rb

```diff
@@ -0,0 +1,43 @@
+class WalksController < ApplicationController
+  def index
+  end
+
+  def new
+    current_user
+    p @current_user
+    @walk = Walk.new
+  end
+
+  def create
+    @user = User.find_by_id(params[:user_id])
+    @walk = Walk.new(walks_params)
+    @walk.user = @current_user
+    @user.walks << @walk
```

Comment:
Is this code setting the user who creates the walk as the requester? I'm not seeing how and that's important in how this works. 

You'd need to change line 14 I think to read `@walk.requester_id = @current_user.id`

Also, where is the @current_user instance variable used on line 14 declared?